flink-libraries/flink-table/src/test/scala/org/apache/flink/table/expressions/UserDefinedScalarFunctionTest.scala

```diff
@@ -89,6 +90,14 @@ class UserDefinedScalarFunctionTest extends ExpressionTestBase {
       "Nullable(f0)",
       "Nullable(f0)",
       "42")
+
+    // test row type input
+    testAllApis(
+      Func19('f11),
+      "Func19(f11)",
+      "Func19(f11)",
+      "12,true,1,2,3"
```

Comment:
is the result of a scalar function automatically flattened?

Reply:
No flattening is happening here. It calls the `toString` method of `Row`. If it would be flattened, then only the first field would be returned by the test base.

Reply:
I see, that makes sense. 

I would be nice to be able to distinguish the nested and flattened case. 
But I that would exceed the scope of this PR.